operations/boot_orchestration/Create_a_Session_Template_to_Boot_Compute_Nodes_with_SBPS.md

```diff
@@ -0,0 +1,185 @@
+
+### `<schema version>`
+
+The version of the SBPS schema. `v1` is the only supported version.
+
+### `<IQN Domain>`
+
+The domain name portion of the iSCSI Qualified Name (IQN).
+
+For example, an IQN might look like this:
+
+```text
+iqn.2023-06.csm.iscsi:ncn-w002
+```
+
+In this example:
+
+- `iqn.2023-06.csm.iscsi` is the IQN domain.
+- `ncn-w002` is a unique identifier for the storage device.
+
+The IQN Domain helps in ensuring unique identification of iSCSI targets within a given namespace, allowing for proper routing and management of storage resources over the network
+
+For SBPS, only the domain portion of the IQN needs to be supplied, not the entire IQN.
+
+### `<DNS Service (SRV) record reference>`
+
+A DNS SRV (Service) record is a type of Domain Name System (DNS) resource record used to specify information about services offered by a particular domain.
+
+Here's how a DNS SRV record is structured:
+
+- Service: Specifies the symbolic name of the service.
+- Protocol: Specifies the transport protocol of the service, such as TCP or UDP.
+- Name: The domain name for which this record is valid.
+- Other elements were omitted for clarity and brevity.
+
+For example, DNS SRV record might look like this:
+
+```text
+_sbps-hsn._tcp.odin.hpc.amslabs.hpecorp.net
+```
+
+In this example,
+
+- `_sbps-hsn` is the symbolic name of the service
+- `_tcp` is the transport protocol
+- `odin.hpc.amslabs.hpecorp.net` is the domain name for which the record is valid. This is the system domain.
+
+### `<client discovery timeout in seconds>`
+
+The timeout, in seconds, for attempting to mount the `netroot` via SBPS.
+
+Can be left as an empty string to use the default value of 300 seconds.
+
+### `<ramroot>`
+
+Indicates that the specified S3 path should be copied to RAM \(`tmpfs`\) and mounted locally instead of persisting as a remote file system mount.
+
+Can be left empty. Any string except `"0"` is interpreted as true.
+
+### Example `rootfs_provider_passthrough`
+
+```text
+rootfs_provider_passthrough=sbps:v1:iqn.2023-06.csm.iscsi:_sbps-hsn._tcp.odin.hpc.amslabs.hpecorp.net:300
+```
+
+## `root=` kernel parameter
+
+BOS will construct the `root=` kernel parameter, which will be used by the node when it boots, based on the `rootfs_provider` and `rootfs_provider_passthrough` values.
+
+For SBPS, BOS supplies a protocol `sbps-s3`, the S3 path to the `rootfs`, and the `etag` value \(if it exists\).
+The rest of the parameters are supplied from the `rootfs_provider_passthrough` values as specified above.
+
+BOS will construct it in the following format:
+
+```text
+root=sbps-s3:s3-path:<etag>:<transport>:<schema version>:<IQN Domain>:<DNS SRV record reference>:<client discovery timeout in seconds>:<ramroot>
+```
+
+### Example kernel parameter
+
+```text
+root=sbps-s3:s3://boot-images/4fab0408-0bfe-4668-b957-964f8ff0e4e9/rootfs:b6ea7a2314d54dead0c94223863b3488-1977:sbps:v1:iqn.2023-06.csm.iscsi:_sbps-hsn._tcp.odin.hpc.amslabs.hpecorp.net:30
```

Comment:
Should the client discovery timeout be 300 instead of 30 in this example for consistency with the default value mentioned above?

Reply:
Checked with COS, 30s is a fine default.

Reply:
Changed.